packages/map-template/src/hooks/useMapBoundsDeterminer.js

```diff
@@ -72,51 +80,132 @@ const useMapBoundsDeterminer = () => {
      */
     function determineMapBounds() {
         const currentVenue = venuesInSolution.find(venue => venue.name.toLowerCase() === currentVenueName.toLowerCase());
+        let desktopPadBottom = null;
+        let desktopPadLeft = null;
+        let mobilePadBottom = null;
+
         if (mapsIndoorsInstance && currentVenue) {
             setMapPositionInvestigating(true);
 
+            // Parse center prop into coordinates. If it is not included in the URL, latLng are undefined.
+            const [latitude, longitude] = center
+                ? center.split(",").map(Number)
+                : [undefined, undefined];
+
+            // Create centerPoint object.
+            const centerPoint = { geometry: { type: 'Point', coordinates: [latitude, longitude] } };
+
+            // Returns Venue that intersects with center prop.
+            const intersectingVenueWithCenterPoint = venuesInSolution.find(venue => {
+                return turf.booleanIntersects(venue.geometry, centerPoint.geometry);
+            })
+
+            // If startZoomLevel is not defined, fallback to 16 default value.
+            const zoomLevel = startZoomLevel ?? 16;
+
+
+            Promise.all([getDesktopPaddingBottom(), getDesktopPaddingLeft(), getMobilePaddingBottom()]).then(([desktopPaddingBottom, desktopPaddingLeft, mobilePaddingBottom]) => {
```

Comment:
Won't we risk that the "pad" variables are null since this is asynchronous and the code following it is immediately executed?